### PR TITLE
first two parts of the assignment

### PR DESCRIPTION
**Pull request by dmitriyrotaenko**

### CODE REVIEW

File: formLogic.js [hunk 1]
```diff
@@ -27,8 +27,27 @@ function clearFields() {
...
+  const filmList = $$("films");
+
+  const filmData = form.getValues();
+
+  if(filmData.id) {
+    filmList.updateItem(filmData.id, filmData);
+    form.clear();
+    form.clearValidation();
+  }
+  else webix.message({type: "error", text: "Something went wrong"});
```
Comment (Zwillinge) [inline formLogic.js]: if there's no id among data fields, you can add it to the datatable.
and after `if..else` block , clear (validation as well) the form

File: formLogic.js [hunk 2]
```diff
@@ -1,17 +1,21 @@
...
-  });
+function saveFilm() {
+  const films = $$("films");
+  const form = this.getFormView();
+  const filmData = form.getValues();
+
+  if(filmData.id) {
+    films.updateItem(filmData.id, filmData);
+  } else {
+    if(form.validate()) {
```
Comment (Zwillinge) [inline formLogic.js]: you need to validate data for both cases, adding and editing

File: formLogic.js [hunk 3]
```diff
@@ -1,17 +1,21 @@
...
+      films.add(filmData);
+      webix.message({type: "success", text: "Film was added successfully"});
+      films.showItem(filmData.id)
```
Comment (Zwillinge) [inline formLogic.js]: id should be taken from from return value of the `.add` method

File: formLogic.js [hunk 4]
```diff
@@ -1,17 +1,21 @@
...
+
+      form.clear();
```
Comment (Zwillinge) [inline formLogic.js]: clearing should be done after both adding and editing

File: table.js [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+function deleteItem(e, id) {
+  webix.confirm({
+    title: "Delete this item?",
+    text: "Item will be deleted",
+    ok: "Yes", cancel: "No"
+  })
+  .then(() => {
+    this.remove(id);
+    if(this.getParentView() === $$("Dashboard")) {
```
Comment (Zwillinge) [inline table.js]: looks strange. why do you need this check?

Comment (dmitriyrotaenko) [reply table.js]: Really strange. Some crazy tests, I suppose :)

File: formLogic.js [hunk 6]
```diff
@@ -1,34 +1,49 @@
...
+  const films = $$("films");
+  const form = this.getFormView();
+  const filmData = form.getValues();
+  
+  const rank = +films.getItem(films.getLastId()).rank + 1; // to display rank correctly
+
+
+
+  if(form.validate()) {
+    switch(filmData.id) {
```
Comment (Zwillinge) [inline formLogic.js]: do not use switch as simple `if..else` to check for the existense of a property 
swicth it usually used when a param can have a 3 > different values

File: formLogic.js [hunk 7]
```diff
@@ -1,34 +1,49 @@
...
+        const id = films.add(filmData);
+        webix.message({type: "success", text: "Film was added successfully"});
+        films.showItem(id);
+      break;
+      default: 
+        films.updateItem(filmData.id, filmData);
+        webix.message({type: "success", text: "Film was updated successfully"});
+    } 
+
+    clearForm(form);
```
Comment (Zwillinge) [inline formLogic.js]: if using a separate function, it is better to make it self-sufficient and get form by id within this function

File: formLogic.js [hunk 8]
```diff
@@ -1,34 +1,49 @@
...
-
   webix.confirm({
     title: "Clear fileds?",
     text: "All fields will be cleared! Are you sure?",
     ok: "Yes", cancel: "No"
   })
   .then(() => {
-    form.clear();
-    form.clearValidation();
+    clearForm(this.getFormView());
```
Comment (Zwillinge) [inline formLogic.js]: the same about form  - get it within the `clearForm` function

**Pull request merged**